pandas/core/index.py

```diff
@@ -971,11 +971,13 @@ def _convert_list_indexer(self, keyarr, kind=None):
 
             if self.inferred_type == 'mixed-integer':
                 indexer = self.get_indexer(keyarr)
+                if not np.all(np.in1d(indexer, self.values)):
```

Comment:
your test doesn't hit that line though it goes thru
can u make a test that is explicitly caught there?